packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -309,21 +309,28 @@ export const createBundle = async ({
                     )
 
                     // If the project is using template extensbility, we must look into the base template's package.json to find the versions of pwa-kit packages
-                    const extendsPkgJsonPath = path.join(extendsTemplate, 'package.json')
+                    const extendsPkgJsonPath = path.join(process.cwd(), 'package-lock.json')
                     if (existsSync(extendsPkgJsonPath)) {
-                        const {dependencies = {}, devDependencies = {}} = JSON.parse(
-                            readFileSync(extendsPkgJsonPath, 'utf-8')
-                        )
-
-                        for (const pkg of pwaKitDeps) {
-                            packageVersions[pkg] = dependencies[pkg] || devDependencies[pkg]
+                        const {packages = {}} = await readJson(extendsPkgJsonPath)
+
+                        for (const pkgKey in packages) {
+                            if (pkgKey.endsWith(extendsTemplate)) {
+                                const pkg = packages[pkgKey]
+                                for (const dep of pwaKitDeps) {
+                                    if (pkg.dependencies[dep]) {
+                                        packageVersions[dep] = pkg.dependencies[dep]
+                                    }
```

Comment:
@wjhsf Wanted to get your feedback on this approach using `package-lock.json` to read pwa kit package versions

Reply:
This is still searching only the direct dependencies of the `extendsTemplate` package. We want to search all deps:
```ts
for (const pkgKey of Object.keys(packages)) {
  for (const pwaKitDep of pwaKitDeps) {
    if (pkgKey.endsWith(`node_modules/${pwaKitDep}`)) {
      packageVersions[pwaKitDep] = packages[pkgKey].version
    }
  }
}
```
The above would get the installed version of each PWA Kit dep, regardless of where it is in the dependency tree. But I think a more robust solution would use `npm ls`, as `npm ls` works for npm projects and yarn projects. (Reading `package-lock.json` only works for npm projects.)